pyflakes/test/test_imports.py

```diff
@@ -564,6 +563,15 @@ def bar():
         fu
         ''', m.UnusedImport, m.UndefinedName)
 
+    def test_shadowedInNestedScope(self):
+        self.flakes('''
+        import fu
+        def bar():
+            import fu
+            fu
+        fu
+        ''', m.ImportShadowedByImport)
```

Comment:
That could work. Though, #44 makes me think of another case:

```python
import foo

class bar:
    import foo

foo
```

Probably there should be no errors here, but right now it looks like the 2nd `import foo` yields "test.py:4: import 'foo' from line 1 shadowed by import"